src/Core/Domain/ValueObject/Reduction.php

```diff
@@ -111,30 +113,20 @@ private function assertIsAllowedType(string $type)
 
     /**
      * @param string $type
-     * @param float $value
+     * @param DecimalNumber $value
      *
      * @throws DomainConstraintException
      */
-    private function assertIsValidValue(string $type, float $value)
+    private function assertIsValidValue(string $type, DecimalNumber $value)
```

Comment:
We should add unit tests on this class to make sure there was no regressions due to these internal changes